### DISCUSSION TITLE
Separating business logic from UI

### DISCUSSION THREAD

**Discussion by omiz** [MobileNativeFoundation/discussions#71] [General]
Hello Everybody!

We at [rocketform](https://www.rocket-es.com) provide claim reporting services on iPad to many insurance companies. The business logic between one of our customers and another would range between identical to some what different in some small or big details.

The use of MVVM is a big help in separating the business logic. We also need a feature manager which adds some complexity to the View Model.

We are looking for moving this logic to a separated framework library to allow the ability to provide a skimmed down app with a smaller size if needed.

The fact that some of our customers has 90% similar logic to another would make some code duplication.

Is there best practices for moving the business logic to frameworks and what is your experience with it?

By the way some of our customers provide the server by their own so Server or Database driven development is not an easy option for us.

**Comment by Exadra37** [comment]
> The fact that some of our customers has 90% similar logic to another would make some code duplicatin.
>
> Is there best practices for moving the business logic to frameworks and what is your experience with it?

In the past I have worked as an API developer at a leading mobile app company in UK, doing mobile apps for the biggest retailers in UK, and the approach we took was to move all the business logic, as possible, to the backend, and made the mobile app as dumb as possible.

This approach allowed us to fix bugs without the need to release new mobile app versions and trigger the force update kill switch.

All the mobile apps were driven by dynamic configuration fetched from the backend, so that we could also tweak its behaviour without releasing a new version.

For the common functionality that couldn't be delegated to the backend an in-house framework was built and used across all apps.

Sorry if cannot give you a more elaborated answer, but I wasn't in the mobile team :)
